### ISSUE TITLE
Cleanly distinguish input map entries with and without modifiers

### ISSUE THREAD

**Issue by tiloc** [godotengine/godot-proposals#9450]
### Describe the project you are working on

A game where I can move a character through the keyboard. The characters walks when I press a directional key (e.g. WASD or cursor keys) and runs, when I additionally hold the shift key.

### Describe the problem or limitation you are having in your project

I had added an entry "walk_forward" (W) and an entry "run_forward" (SHIFT+W) into the input map. In my _process_physics method, I tried to query the input with Input.get_action_strength. If I only hold W, then only "walk_forward" has strength 1, but when I hit SHIFT+W, both walk_forward and run_forward have strength 1.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allow entries into the input map that only trigger when the exact modifiers as specified are (and especially: are not) pressed.

My situation seems very similar to the proposal described here: https://github.com/godotengine/godot-proposals/issues/2464
However, this other issue is allegedly fixed since 4.0, but I am experiencing my issue on Godot 4.2.1. Not sure if I am experiencing a bug, or the original proposal only fixed a particular situation and mine is different.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I can imagine two possible ways:
- make all matches precise, incl. pressed and unpressed modifiers
- Allow ordering the entries in the input map and let Godot search the map until it has the first match incl. all pressed modifiers (i.e. I would put the actions with more pressed modifiers first into the map)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I am currently working around this by hard-coding modifier matching into my code, but feel that input matching should be declarative to the map and not handled half by the input map and half by extra code. If I ever allow players to define their own input mapping then my hard-coded logic would be very much in the way (e.g. if they can define Tab + Shift/Tab to cycle through weapons).

### Is there a reason why this should be core and not an add-on in the asset library?

I'm not familiar enough with the details of Godot to know whether this could even be done through an asset library. Subjectively I feel that input mapping is a core functionality of a gaming engine.

**Comment by KoBeWi** [comment]
Did you try `exact_match` argument?

**Comment by tiloc** [comment]
> Did you try `exact_match` argument?

Duh, I missed that this existed. For get_action_strength this will hopefully work, I'll try it. For Input.get_vector it doesn't seem to exist, I am also using that sometimes, but might be able to convert to get_action_strength if needs be.

**Comment by Calinou** [comment]
> I had added an entry "walk_forward" (W) and an entry "run_forward" (SHIFT+W) into the input map.

Out of curiosity, why not just have a `walk_forward` action and a `run` action? In your code, make the `run` action only change the player speed while `walk_forward` is held. This is how most games handle sprinting in my experience.

**Comment by AThousandShips** [comment]
It's very unusual to have run and walk bound to *different* keys, I'd even say no game should ever do that, so as Calinou suggests it should be handled by a separate key, especially if you want controller support etc., having `W` for walk forward and, idk, `T` for run forward makes no sense, so you can just break it into a `run` action, and now you can make `run` the `B` button on your controller which you couldn't do otherwise

**Comment by tiloc** [comment]
> > I had added an entry "walk_forward" (W) and an entry "run_forward" (SHIFT+W) into the input map.
> 
> Out of curiosity, why not just have a `walk_forward` action and a `run` action? In your code, make the `run` action only change the player speed while `walk_forward` is held. This is how most games handle sprinting in my experience.

I have been able to do this now. I didn't realise I could have an entry in the input map that was only a modifier. After playing my PS5 and my Switch for a while and taking note how various games sprint, I have been able to imitate all common modes of sprinting with existing Godot 4.2.1.
1. Holding Shift in addition to the movement keys (as done in some PC games like Hell Divers 2)
2. Toggling sprinting through a button or L3 (done in many console games)
3. Holding a controller trigger in addition to stick movement (done in many console games)
